packages/discord.js/src/structures/GuildChannel.js

```diff
@@ -145,8 +147,20 @@ class GuildChannel extends BaseChannel {
    * @readonly
    */
   get position() {
-    const sorted = this.guild._sortedChannels(this);
-    return [...sorted.values()].indexOf(sorted.get(this.id));
+    const selfIsCategory = this.type === ChannelType.GuildCategory;
+    const types = getSortableGroupTypes(this.type);
+
+    let count = 0;
+    for (const channel of this.guild.channels.cache.values()) {
+      if (!types.includes(channel.type) || !(selfIsCategory || channel.parentId === this.parentId)) continue;
```

Comment:
```suggestion
      if (!(types.includes(channel.type) && (selfIsCategory || channel.parentId === this.parentId))) continue;
```

well a little bit better

Reply:
IMO this reads slightly better, but the current code is fine as well

```suggestion
      if (!types.includes(channel.type) || (!selfIsCategory && channel.parentId !== this.parentId)) continue;
```

Reply:
not too sure about the performance of yours
I have tested mine and it it is 25% better

Reply:
I think you may need to do more runs, 25% is an impressive performance gap after all. I'll also see if I can open source the benchmark suite I'm using, which uses `Histogram` and `performance.timerify`.

Reply:
Well only the if is 25% better lol
I am not speaking of the whole function
> btw nice that you will publish it

Reply:
https://github.com/discordjs/discord.js/pull/8393#discussion_r933528705

Reply:
okey then take almeidx one

Reply:
I ran the benchmarks, it seems @almeidx's is the fastest after all, testing with my support guild's channels (38 in total with multiple categories):

![image](https://github.com/discordjs/discord.js/assets/24852502/60d2a0e5-720c-4880-a8a4-9961e1d642ae)

However, it's important to note that this performance varies from the case-to-case.